### PR TITLE
Add unit tests for sockets wrapper

### PR DESCRIPTION
**Pull request by yourslab**
Regular expressions are used to massage the header file into something that CMock can parse.
If that doesn't work, then I create a header file with all the functions I want to mock and use that instead.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by yourslab** [comment]
I am going to add docstrings

File: platform/posix/transport/utest/sockets_utest.c [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+
+#define NUM_ADDR_INFO        3
+#define SEND_RECV_TIMEOUT    0
+#define HOSTNAME             "amazon.com"
+#define PORT                 80
+
+static struct addrinfo * addrInfo;
+static ServerInfo_t serverInfo;
+
+static void allocateAddrInfoLinkedList( struct addrinfo ** head )
```
Comment (sarenameas) [inline platform/posix/transport/utest/sockets_utest.c:24]: Could you please document your static private functions.

File: platform/posix/transport/utest/sockets_utest.c [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+    int tcpSocket = 1;
+
+    shutdown_ExpectAnyArgsAndReturn( 0 );
+    close_ExpectAnyArgsAndReturn( 0 );
+
+    socketStatus = Sockets_Disconnect( tcpSocket );
+    TEST_ASSERT_EQUAL( SOCKETS_SUCCESS, socketStatus );
+}
+
+void test_Sockets_Disconnect_Invalid_Socket( void )
```
Comment (sarenameas) [inline platform/posix/transport/utest/sockets_utest.c:168]: Please document your tests

File: platform/posix/transport/utest/sockets_utest.c [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+    int tcpSocket = 1;
+
+    /* Passing a NULL socket should fail. */
+    socketStatus = Sockets_Connect( NULL,
+                                    &serverInfo,
+                                    SEND_RECV_TIMEOUT,
+                                    SEND_RECV_TIMEOUT );
+    TEST_ASSERT_EQUAL( SOCKETS_INVALID_PARAMETER, socketStatus );
+
+    /* Passing a hostName should fail. */
```
Comment (sarenameas) [inline platform/posix/transport/utest/sockets_utest.c:189]: Passing a NULL hostName should fail?

File: platform/posix/transport/utest/sockets_utest.c [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+void test_Sockets_Disconnect_Invalid_Socket( void )
+{
+    SocketStatus_t socketStatus;
+    int tcpSocket = 0;
+
+    socketStatus = Sockets_Disconnect( tcpSocket );
+    TEST_ASSERT_EQUAL( SOCKETS_INVALID_PARAMETER, socketStatus );
+}
+
+void test_Sockets_Connect_Invalid_Params( void )
```
Comment (sarenameas) [inline platform/posix/transport/utest/sockets_utest.c:177]: I think we also need a NULL pServerInfo parameter test?

**Comment by yourslab** [comment]
Closing because #1172 contains what is in this PR

**Pull request closed without merging**